omf-python/src/file/reader.rs

```diff
@@ -250,4 +250,22 @@ impl PyReader {
             .collect::<Result<Vec<_>, _>>()
             .map_err(|e| PyErr::new::<PyValueError, _>(e.to_string()))
     }
+
+    /// Read a Boundary array.
+    pub fn array_boundaries(&self, array: &PyBoundaryArray) -> PyResult<Vec<f64>> {
+        let numbers_f64 = self
+            .0
+            .array_boundaries(&array.0)
+            .map_err(|e| PyErr::new::<PyIOError, _>(e.to_string()))?
+            .try_into_f64()
+            .map_err(|e| PyErr::new::<PyIOError, _>(e.to_string()))?;
```

Comment:
is this needed twice?

Reply:
I probably need to change all these to convert to OmfException so can leave it like this.

Reply:
I _think_ so, but may be wrong:

 - `pub fn array_boundaries(&self, array: &Array<array_type::Boundary>) -> Result<Boundaries, Error>` this can return an error, 
 - and so can `pub fn try_into_f64(self) -> Result<BoundariesF64, Error>`

so I assumed they needed to be handled after each call?

Reply:
It looks like its not mapping to an error it throws, so it'll will throw a different error, but I think most of our current error code is doing this so its ok I will need to update it anyway to create an OmfException.

Reply:
Searching GitHub for similar code... looks like we can simplify the lines to:

```
.map_err(|e| PyIOError::new_err(e.to_string()))?
```

but that would need to be done across the board, and also doesn't reduce the need for two instances of `.map_err()`